docs/index.bs

```diff
@@ -3481,93 +3452,79 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
-              1. If |cachedURL| [=url/equals=] |requestURL| with the *exclude fragments flag* set, then:
-                  1. Add |record|[0] to |requestArray|.
-                  1. Add |record|[1] to |responseArray|.
+      1. Let |storage| be null.
+      1. If the optional argument |targetStorage| is omitted, set |storage| to the [=context object=]'s [=request response list=].
+      1. Else, set |storage| to |targetStorage|.
+      1. [=list/For each=] |item| of |storage|:
+          1. Set |cachedURL| to |item|'s request's associated [=Request/request=]'s [=request/url=].
+          1. Set |requestURL| to |request|'s associated [=Request/request=]'s [=request/url=].
+          1. If |options|.{{CacheQueryOptions/ignoreSearch}} is true, then:
+              1. Set |cachedURL|'s [=url/query=] to the empty string.
+              1. Set |requestURL|'s [=url/query=] to the empty string.
+          1. If |cachedURL| [=url/equals=] |requestURL| with the *exclude fragments flag* set, then:
+              1. Add a copy of |item|'s request to |requestArray|.
+              1. Add a copy of |item|'s response to |responseArray|.
       1. For each |cachedResponse| in |responseArray| with the index |index|:
           1. Let |cachedRequest| be the |index|th element in |requestArray|.
           1. If |cachedResponse|'s [=Response/response=]'s [=response/header list=] contains no <a>header</a> [=header/named=] \`<code>Vary</code>\`, or |options|.{{CacheQueryOptions/ignoreVary}} is true, then:
-              1. Add an array [|cachedRequest|, |cachedResponse|] to |resultArray|.
+              1. [=list/Append=] |cachedRequest|/|cachedResponse| to |resultList|.
               1. Continue to the next iteration of the loop.
           1. Let |varyHeaders| be the array containing the elements corresponding to the [=http/field-values=] of the <a>Vary</a> header.
           1. Let |matchFailed| be false.
           1. For each |f| in |varyHeaders|:
               1. If |f| matches "<code>*</code>", or the result of running |cachedRequest|.{{Request/headers}} object's {{Headers/get(name)}} method with |f| as the argument does not match the result of running |request|.{{Request/headers}} object's {{Headers/get(name)}} method with |f| as the argument, then:
                   1. Set |matchFailed| to true.
                   1. Break the loop.
-          1. If |matchFailed| is false, add an array [|cachedRequest|, |cachedResponse|] to |resultArray|.
-      1. Return |resultArray|.
+          1. If |matchFailed| is false, [=list/append=] |cachedRequest|/|cachedResponse| to |resultList|.
+      1. Return |resultList|.
   </section>
 
   <section algorithm>
     <h3 id="batch-cache-operations-algorithm"><dfn>Batch Cache Operations</dfn></h3>
 
       : Input
-      :: |operations|, an array of  {{CacheBatchOperation}} dictionary objects
+      :: |operations|, a [=list=] of {{CacheBatchOperation}} dictionary objects
```

Comment:
We never expose this, so it doesn't need to be a dictionary right?

Reply:
Yes. I replaced it with a struct.